### PR TITLE
Feature/issue 208  better build

### LINKED ISSUE #208: Use "Maven Friendly Version"
By using the "[Maven Friendly Version](https://maven.apache.org/maven-ci-friendly.html)" guide, there is no more `pom.xml` file modification needed in Git for a release.

And next, do the same on other languages plugins

### PR DESCRIPTION
**Pull request by jycr**
fix #208 

* [Use Maven CI Friendly Versions](https://github.com/green-code-initiative/ecoCode/pull/213/commits/96d56020e66dc274ba2f1d420382c2522a0ec501) and [removes scripts for release management](https://github.com/green-code-initiative/ecoCode/pull/213/commits/63fb3b4cf0ff54a960063a2df0460adbd966a8e8). The release process is now much simpler: you just need to do a build on a tag checkout (the `pom.xml` files do not need to be modified anymore. So there is no need to create commits just for the version change in the `pom.xml`). See [Maven CI Friendly Versions - Apache Maven Project](https://maven.apache.org/maven-ci-friendly.html) and [Maven Git Versioning Extension](https://github.com/qoomon/maven-git-versioning-extension) for more information
* [Ensure build reproducibility](https://github.com/green-code-initiative/ecoCode/pull/213/commits/8464651d7e9729d0e6109c3b009cd575f95a2b9c). See [Configuring for Reproducible Builds - Apache Maven Project](https://maven.apache.org/guides/mini/guide-reproducible-builds.html) for more information
* [Adds Maven Wrapper](https://github.com/green-code-initiative/ecoCode/pull/213/commits/d8356fdba3f78ad575a4c440d4e34043d610c9c4) and [migrates scripts to use Maven Wrapper](https://github.com/green-code-initiative/ecoCode/pull/213/commits/7fd4d8da3cce9d55f7fa5b78685a0a8c4f78185b) (to ensure consistency of Maven version used across different environments). See [ecoCode#336](https://github.com/green-code-initiative/ecoCode/issues/336) and Apaceh Maven Wrapper documentation](https://maven.apache.org/wrapper/) for more information.
* [Updates asciidoctor-maven-plugin to 3.0.0](https://github.com/green-code-initiative/ecoCode/pull/213/commits/58581808b1b3ac7a7be798f8c4c26511da6048f2)

### CODE REVIEW

**Comment by dedece35** [comment]
Hi @jycr,
sorry but we have to discuss about this feature because I don't know it and I have some questions.

**Comment by jycr** [comment]
> Hi @jycr, sorry but we have to discuss about this feature because I don't know it and I have some questions.

No problem to discuss it together.
But we can wait to split `java-plugin` into its own repo before finishing this PR?

**Comment by dedece35** [comment]
> > Hi @jycr, sorry but we have to discuss about this feature because I don't know it and I have some questions.
> 
> @dedece35: No problem to discuss it together. But we can wait to split `java-plugin` into its own repo before finishing this PR?

@jycr ... yes I think we should wait automatic release PR before merging this one

**Review by dedece35 — DISMISSED** [review]

**Review by dedece35 — CHANGES_REQUESTED** [review]

**Review by dedece35 — CHANGES_REQUESTED** [review]
add an update in CHANGELOG.md, please

File: .mvn/extensions.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<extensions xmlns="http://maven.apache.org/EXTENSIONS/1.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/EXTENSIONS/1.0.0 https://maven.apache.org/xsd/core-extensions-1.0.0.xsd">
+	<extension>
+		<groupId>me.qoomon</groupId>
+		<artifactId>maven-git-versioning-extension</artifactId>
+		<version>9.6.5</version>
```
[the remark this answers is not in the document]
Comment (jycr) [reply .mvn/extensions.xml]: I have updated all Maven dependencies/plugins

**Comment by jycr** [comment]
> add an update in CHANGELOG.md, please

I [have added](https://github.com/green-code-initiative/ecoCode/pull/213/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR14-R16):

> [#208](https://github.com/green-code-initiative/ecoCode/issues/208) Improves build process with [Maven Wrapper](https://maven.apache.org/wrapper/) and simplifies release management (now, just need to do a build on a tag checkout).
> If you have difficulties on some IDEs (like IntelliJ IDEA), you can follow the procedure: [configuration for Multi Modules Project](https://github.com/qoomon/maven-git-versioning-extension?tab=readme-ov-file#intellij---multi-modules-projects)

Do you need me to add any further details?

**Review by dedece35 — APPROVED** [review]

**Pull request merged**